ping.go

```diff
@@ -624,7 +637,7 @@ func (p *Pinger) processPacket(recv *packet) error {
 	var m *icmp.Message
 	var err error
 	if m, err = icmp.ParseMessage(proto, recv.bytes); err != nil {
-		return fmt.Errorf("error parsing icmp message: %w", err)
+		return fmt.Errorf("%w: %s", errParseICMPMessage, err)
```

Comment:
This should probably be `"%w: %w"`.

Reply:
You can only wrap with a single error. If you try with 2:

> fmt.Errorf call has more than one error-wrapping directive %wprintf

The idea is that functions calling this unexported function can check what block of code failed by using `errors.Is(err, errParseICMPMessage)`, without relying on errors defined in the `icmp` package (if there is any 🤔). We keep the original error as a string for the user to see in logs etc.

Reply:
Makes sense.